viewing_party/party.py

```diff
@@ -1,23 +1,158 @@
 # ------------- WAVE 1 --------------------
 
+import re
+
+
 def create_movie(title, genre, rating):
-    pass
+    
+    if title and genre and rating:
+        return {
+            "title": title,
+            "genre": genre,
+            "rating": rating,
+        }
+    else:
+        return None    
+
+def add_to_watched(user_data,movie):
+
+    user_data['watched'].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie): 
+
+    user_data["watchlist"].append(movie) 
+    return user_data 
+
+def watch_movie(user_data,title):
+
+    for movie in user_data["watchlist"]:
+        if movie['title'] == title:
+            user_data["watched"].append(movie)
+
+            user_data["watchlist"].remove(movie)    
+            break
+    return user_data    
+
+           
+               
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    if user_data["watched"] == []:
+
+        return 0.0
+    else:    
+        sum = 0
+        for movie in user_data["watched"]:
+            sum += movie['rating']
+
+        avg = sum / len(user_data["watched"]) 
+        return avg 
+
+def get_most_watched_genre(user_data):  
+    
+
+    if user_data["watched"] ==[]:
+         return None
+
+    genre_freq ={}
+    for movie in user_data["watched"]:
+        if movie['genre'] in genre_freq:
+            genre_freq[movie["genre"]] += 1
+        else:
+            genre_freq[movie["genre"]] = 1
+    
+
+    for genre, value in genre_freq.items():
+        if value == max(genre_freq.values()):
+            return genre 
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):

```

Comment:
Great job! One thing to consider: it might be a good idea to have a helper function that gets all of the movies watched by friends with duplicates removed. That would make the code here a little cleaner since you would not have to loop through each friend for every movie the user watched. If you had a list of `friends_movies`, you can replace lines 82 - 89 with
```
if movie not in friends_movie:
    unique_watched.append(movie)
```
This helper function would also be very useful for the next function which also needs all of the movies (without duplicates) that the friends have seen.